core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -213,6 +217,14 @@ object ConsoleConsumer extends Logging {
       .withRequiredArg
       .describedAs("blacklist")
       .ofType(classOf[String])
+    val partitionIdOpt = parser.accepts("partition", "The partition to consume from.")
+      .withRequiredArg
+      .describedAs("partition")
+      .ofType(classOf[java.lang.Integer])
+    val offsetOpt = parser.accepts("offset", "The offset id to consume from, default to -2 which means from beginning; while value -1 means from end")
```

Comment:
Two questions:

1. If this has a default, shouldn't it have a `defaultsTo` call?
2. Do we want the magic constants `-1` and `-2` to leak into this public interface? Maybe we could make this accept a `String` which is then either parsed to `earliest`, `latest`, or an `Long`?